### ISSUE TITLE
SetBasePath method in ApiClient should be marked as obsolete

### ISSUE THREAD

**Issue by deankinane** [docusign/docusign-csharp-client#274]
The SetBasePath method is marked as obsolete in every class except the ApiClient class.
It's current implementation also results in unintended behaviour:

```C#
var apiClient = new ApiClient();
apiClient.SetBasePath(Demo_REST_BasePath); // RestClient path is not updated
apiClient.SetOAuthBasePath(); // OAuth path is now updated based on new base path
```

This results in an ApiClient which is pointing at production for the Rest API and demo for the OAuth API.

Or another, more realistic use case:
```C#
apiClient.SetBasePath(account.BaseUri); // Has no effect on RestClient
```

The above will seemingly work for US accounts (as the account base path is likely the same) but will then fail if an EU users authenticates.

I suggest removing the implementation leaving only a stub and marking as obsolete for consistency across the repo.

**Comment by nrobert** [comment]
Agree on that, the `SetBasePath` method is misleading for developers

**Comment by LarryKlugerDS** [comment]
Good news: Engineering is working on this issue. Will be fixed in the next release.

**Comment by RajRele** [comment]
This has been resolved with v5.1.0 for v2.1 and v3.7.0 for v2
